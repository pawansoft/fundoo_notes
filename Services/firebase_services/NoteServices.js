import AsyncStorage from '@react-native-async-storage/async-storage';
import Firebase from '../../config/Firebase';

class FirebaseService{
    _storeNoteService = async (notekey, title, note) =>{
        const userid = JSON.parse(await AsyncStorage.getItem('userId')) 
        return new Promise((resolve, reject) => {
            const notes = {
                title : title,
                note : note,
                isDeleted: false
            }
            Firebase.database().ref('Notes/' +userid+'/' +notekey).set({
                NotesDetail : notes
            }).then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

    _getNoteService = async() => {
        const userid = JSON.parse(await AsyncStorage.getItem('userId'));
       return new Promise((resolve, reject)=> {
        Firebase.database()
        .ref('Notes/' +userid)
        .once('value', snapshot => {
            resolve(snapshot.val())
        }).catch(error => reject(error))
       })
    }

    _cleanRecycleBinService = async() => {
        const userid = JSON.parse(await AsyncStorage.getItem('userId'));
        return new Promise((resolve, reject) => {
            Firebase.database()
            .ref('Notes/' +userid)
        })
    }

    _updateNoteService = async(key, title, note) => {
        const userid = JSON.parse(await AsyncStorage.getItem('userId'));
        return new Promise((resolve, reject) => {
            const notes = {
                title : title,
                note : note,
                isDeleted : false
            }
            Firebase.database().ref('Notes/'+userid+ '/'+key).set({
                NotesDetail : notes
            })
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

    _deleteNotesService = async(key, title, note) => {
        const userid = JSON.parse(await AsyncStorage.getItem('userId'))
        return new Promise((resolve, reject) => {
            const notes = {
                title : title,
                note : note,
                isDeleted : true
            }
            Firebase.database().ref('Notes/'+userid+ '/'+ key).set({
                NotesDetail : notes
            })
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

//Level function start from here 
    _addLevelService = async (label) => {
        const userid = JSON.parse(await AsyncStorage.getItem('userId'))
        return new Promise((resolve, reject) => {
            Firebase.database().ref('Labels/'+userid).push({
                label : label
            })
            .then(() => resolve('success'))
            .catch(error => reject (error))
        })
    }
    
    _getLevelService = async() => {
        const userid = JSON.parse(await AsyncStorage.getItem('userId'));
        return new Promise((resolve, reject)=> {
         Firebase.database()
         .ref('Labels/'+userid)
         .once('value', snapshot => {
             resolve(snapshot.val())
         }).catch(error => reject(error))
        })
    }

    _restoreNoteService = async(key, title, note) => {
        const userid = JSON.parse(AsyncStorage.getItem('userId'))
        return new Promise((resolve, reject) => {
            const notes = {
                title : title,
                note : note,
                isDeleted : false
            }
            Firebase.database().ref('Notes/'+userid+ '/'+ key).set({
                NotesDetail : notes
            })
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

    _deleteOneNoteService = async(key) => {
        const userid = JSON.parse(await AsyncStorage.getItem('userId'))
        return new Promise((resolve, reject) => {
            let dataRef = Firebase.database().ref('Notes/'+userid+ '/'+key)
            dataRef.remove()
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }
}

export default new FirebaseService();