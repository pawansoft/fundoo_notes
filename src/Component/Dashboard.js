import React,{ Component } from 'react';
import {
    View,
    Text
} from 'react-native'

export default class Dashboard extends Component{
    render(){
        return(
            <View>
                <Text>
                    Welcome to dashboard
                </Text>
            </View>
        )
    }
}