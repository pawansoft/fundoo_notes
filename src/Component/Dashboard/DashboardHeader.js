import AsyncStorage from '@react-native-async-storage/async-storage';
import React, { Component } from 'react';
import { TouchableOpacity } from 'react-native';
import RNFetchBlob from 'react-native-fetch-blob'
// import RNFetchBlob from 'react-native-fetch-blob';
import { Appbar, Avatar, Searchbar} from 'react-native-paper';
import Firebase from '../../../config/Firebase';
import dashboardStyle from '../../Style/dashboardStyle';

// const Blob = RNFetchBlob.polyfill.Blob
// const fs = RNFetchBlob.fs
// window.XMLHttpRequest = RNFetchBlob.polyfill.XMLHttpRequest
// window.Blob = Blob
// const Fetch = RNFetchBlob.polyfill.Fetch

export default class DashboardHeader extends Component{
    constructor(props) {
        super(props) 
        this.state ={
            imageUri: ''
        } 
    }

    componentDidMount = async() => {
       await this.getProfileImageService()
        .then(uri =>{
            console.log(uri);
            this.setState({
                imageUri: uri                
            })
        }).catch(error => console.log(error)) 
    }

    getProfileImageService =() => {
        return new Promise(async (resolve, reject) => {
            const userid = await AsyncStorage.getItem('userId');
            Firebase.storage().ref('/' +userid).getDownloadURL()
            .then(url => resolve(url))
            .catch(error => reject(error))
        })
    }

    render(){
        return(
            <Appbar style = {dashboardStyle.headerContainer}>
                <Appbar.Action
                icon = 'menu'
                onPress = {() => this.props.navigation.openDrawer()}/>
                
                <Searchbar
                
                placeholder = 'Search Notes' 
                style = {{width : '60%', color: 'black', backgroundColor: 'grey'}}
                onPress = {this.selectView}/>
                
                <Appbar.Action
                    style = {{marginRight : 10}}
                    icon = {(this.props.listView) ? 'view-grid-outline' : 'view-agenda-outline'}
                    onPress={this.props.onPress}/>
                <TouchableOpacity
                onPress = {this.props.onSelectProfile}>
                <Avatar.Image style = {{flexDirection: "row", justifyContent: "space-between", alignItems: "center"  }} 
                size = {40} 
                source = {(this.state.imageUri == '') ?require('../../assets/Profile.png'): {uri : this.state.fileUri}}/>
                </TouchableOpacity>
            </Appbar>
        )
    }
}