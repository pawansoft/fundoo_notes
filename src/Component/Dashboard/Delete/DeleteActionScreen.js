import React, { Component } from 'react';
import {
    Image,
    ScrollView,
    Text,
    View
} from 'react-native';
import { Appbar, Button, Menu, Snackbar } from 'react-native-paper';
import RBSheet from 'react-native-raw-bottom-sheet';
import AsyncStorage from '@react-native-async-storage/async-storage';
import NotesHolderStyle from '../../../Style/NotesHolderStyle';
import NotesServiceController from '../../../../Services/data_flow_controller/NotesServiceController';
import backgroundImageStyle from '../../../Style/backgroundImageStyle';
import SQLiteLabelServices from '../../../../Services/SQLite_service/SQLiteLabelServices';
import NotesContainerStyle from '../../../Style/NotesContainerStyle';
import Datalayr from '../../../../Services/Datalayer/Datalayr';

export class DeleteActionScreen extends Component {
    constructor(props) {
        super(props)
        this.state = {
            title: '',
            note: '',
            key: '',
            userid: '',
            isDeletable: false,
            selectedLabel: [],
            labelDetails: []
        }
    }

    componentDidMount = async () => {
        const userId = await AsyncStorage.getItem('userId')
        await this.setState({
            userid: userId
        })

        if (this.props.route.params != undefined) {
            console.log(this.state.key);
            await this.setState({
                key: this.props.route.params.notes.NoteKey,
                note: this.props.route.params.notes.Notes,
                title: this.props.route.params.notes.Title,
                selectedLabel: this.props.route.params.notes.Labels
            })
        }

        await SQLiteLabelServices.selectLabelFromSQliteStorage()
            .then(async result => {
                var temp = [];
                if (result.rows.length != 0) {
                    for (let i = 0; i < result.rows.length; ++i)
                        temp.push(result.rows.item(i));
                    this.setState({
                        labelDetails: temp
                    })
                }
            })
    }

    handleRBSheetOpenButton = async () => {
        this.RBSheet.open();
    }

    handleRestoreService = () => {
        Datalayr.handleRestoreNote(this.props.route.params.notes.NoteKey)
        this.props.navigation.push('Home', { screen: 'Delete' })
    }

    handleDeleteNoteForEver = () => {
        this.handleCancel();
        Datalayr.DeleteNoteService(this.state.key)
        this.props.navigation.push('Home', { screen: 'Delete' })

    }

    handleRBSheetOpenButton = async () => {
        this.RBSheet.open();
    }

    onDismissSnakbarHandler = async () => {
        await this.setState({
            isDeletable: false
        })
    }

    handleOpenSnakBar = async () => {
        await this.setState({
            isDeletable: true
        })
    }

    restoreNotesHandler = () => {
        this.handleCancel();
        Datalayr.handleRestoreNote(this.props.route.params.notes.NoteKey)
        this.props.navigation.push('Home', { screen: 'Notes' })
    }
    handleCancel = () => {
        const { onPress } = this.props
        this.RBSheet.close()
    }

    render() {
        return (
            <View style={{ flex: 1 }}>
                <Image style={backgroundImageStyle.backgroundImage} source={require('../../../assets/background1.jpg')}>
                </Image>
                <Appbar style={{ backgroundColor: 'white', width: '20%' }}>
                    <Appbar.Action
                        icon='keyboard-backspace'
                        onPress={() => this.props.navigation.push('Home', { screen: 'Delete' })}
                    />
                </Appbar>
                <ScrollView>
                    <View style={NotesHolderStyle.Title}>
                        <Text style={NotesHolderStyle.TitleFont}
                            onPress={this.handleOpenSnakBar}>
                            {this.state.title}
                        </Text>
                    </View>
                    <View style={NotesHolderStyle.Title}>
                        <Text>
                            {this.state.note}
                        </Text>
                    </View>
                    <View style={NotesHolderStyle.label_text_container}>
                        {(this.state.selectedLabel != undefined) ?
                            this.state.labelDetails.map(data =>
                                this.state.selectedLabel.includes(data.label_id) ?
                                    <React.Fragment key={data.label_id}>
                                        <Text style={NotesHolderStyle.label_text}>{data.label}</Text>
                                    </React.Fragment>
                                    : null

                            ) : null}
                    </View>
                </ScrollView>
                <View>
                    <Appbar style={{ backgroundColor: 'white', justifyContent: 'space-around' }}>
                        <Appbar.Action
                            icon='plus-box-outline' />
                        <Appbar.Action
                            title='Deleted note' />
                        <Appbar.Action
                            icon='dots-vertical'
                            onPress={this.handleRBSheetOpenButton} />
                    </Appbar>
                </View>
                <RBSheet
                    ref={ref => { this.RBSheet = ref }}
                    height={200}
                    customStyles={{
                        container: {
                            borderTopWidth: 1,
                            borderColor: "#d3d3d3",
                            marginBottom: 50
                        },
                        wrapper: {
                            backgroundColor: "transparent",
                        },
                    }}>
                    <View style={{ marginTop: '10%' }}>
                        <Menu.Item
                            icon='autorenew'
                            onPress={this.handleRestoreService}
                            title=' Restore' />
                        <Menu.Item
                            title='Delete forever'
                            icon='delete-outline'
                            onPress={this.handleDeleteNoteForEver} />

                    </View>
                </RBSheet>
                <Snackbar
                    style={{ marginBottom: '30%', flexDirection: 'row', justifyContent: 'space-around', backgroundColor: 'grey' }}

                    visible={this.state.isDeletable}
                    onDismiss={this.onDismissSnakbarHandler}
                    duration={5000}>
                    <View style={{ flexDirection: 'row', justifyContent: 'space-around' }}>
                        <View>
                            <Text style={{ marginTop: 10, color: 'white' }}>
                                Note Deleted Successfully
                            </Text>
                        </View>

                        <View style={{ marginLeft: 50 }}>
                            <Button
                                onPress={this.restoreNotesHandler}>
                                <Text style={{ color: '#cca300' }}>Restore</Text>
                            </Button>
                        </View>
                    </View>
                </Snackbar>
            </View>
        )
    }
}