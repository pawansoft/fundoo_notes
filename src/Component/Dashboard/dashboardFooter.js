import React, { Component } from 'react'
import {View} from 'react-native';
import { Appbar} from 'react-native-paper';
import dashboardStyle from '../../Style/dashboardStyle';

export default class BottomBar extends Component{
    constructor(props){
        super(props)
    }

    handleNewNoteCreateButton = () => {
        this.props.navigation.navigate('NewNotes')
    }

    render(){
        return(
            <View>
                <Appbar style = {dashboardStyle.headerContainer}>
                    <Appbar.Action
                        icon = 'check-box-outline'
                        onPress={() => console.log('Pressed button')}/>
                    
                    <Appbar.Action
                        icon = 'brush'
                        onPress={() => console.log('Pressed button')}/>
                    
                    <Appbar.Action
                        icon = 'microphone-outline'
                        onPress = {() => console.log('Nothing is added so far')}/>
        
                    <Appbar.Action
                        icon = 'panorama'
                        onPress = {() => console.log('Nothing is added so far')}/>
                        
                    <Appbar.Action
                        style = {dashboardStyle.plusButton}
                        icon = {require ('../../assets/plus.png')}
                        onPress = {this.handleNewNoteCreateButton}/>
                </Appbar>
            </View>
        )
    }
}