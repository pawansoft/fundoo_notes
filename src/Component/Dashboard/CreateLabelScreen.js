import React, { Component } from 'react'
import { 
    Text, View, TouchableOpacity, TextInput, ScrollView
} from 'react-native';

import {Appbar, Card, Title} from 'react-native-paper'
import NoteServices from '../../../Services/firebase_services/NoteServices';
import CreateNewLabelStyles from '../../Style/CreateNewLabelStyle';

export default class CreateLabelScreen extends Component{
    constructor(props){
        super(props)
        this.state = {
            label_detail: [],
            labelTextBoxStatus: true,
            labelText: ''
        }
    }


    createLabel = () => {
        this.setState({
            labelTextBoxStatus: !this.state.labelTextBoxStatus
        })
        if(this.state.labelText != '') {
            NoteServices._addLevelService(this.state.labelText).then
            (() => this.props.navigation.push('Home', {screen: 'Label'}))
        }
    }

    handleText = (text) => {
        this.setState({
            labelText: text
        })
    }

    handleBackButton = () => {
        this.props.navigation.push('Home', {screen : 'Notes'})
    }

    cancelLabel = () => {
        this.setState({
            labelText: '',
            labelTextBoxStatus: !this.state.labelTextBoxStatus
        })
        console.log('canceled');
    }
    componentDidMount = async() => {
        await NoteServices._getLevelService().then(async (data) => {           
            let labe_detail = data ? data : {}
            this.setState({
                label_detail : labe_detail
            })
        })
        .catch(error => console.log(error))
        console.log(this.state.label_detail);
    }

    changeToCreateLabel = () => {
        this.setState({
            labelTextBoxStatus: !this.state.labelTextBoxStatus
        })
    }

    render(){
        let labelKey = Object.keys(this.state.label_detail);
        return(
            <View style = {CreateNewLabelStyles.container}>
                <Appbar style = {CreateNewLabelStyles.appbar_Style}>
                    <Appbar.Action
                    icon = "keyboard-backspace"
                    onPress= {this.handleBackButton}/>
                    
                    <Text style = {CreateNewLabelStyles.edit_LabelStyle}>
                        Edit labels
                    </Text>
                </Appbar>

                <Appbar style = {(this.state.labelTextBoxStatus) ? CreateNewLabelStyles.appbar_Style_Active : CreateNewLabelStyles.appbar_Style}>
                    <Appbar.Action
                         icon = {(this.state.labelTextBoxStatus) ? "close" : "plus"}
                         onPress= {(this.state.labelTextBoxStatus) ? this.cancelLabel : this.changeToCreateLabel}
                         style = {{marginRight: '10%'}}
                       />
                       {this.state.labelTextBoxStatus ? 
                        <TextInput
                        placeholder = {'Create new Label'}
                        style = {{width: '60%'}}
                        autoFocus = {this.state.labelTextBoxStatus}
                        onChangeText = {this.handleText}/>
                       :
                       <TouchableOpacity 
                       onPress = {this.createLabel} 
                       style = {{width: '60%'}}>
                           <Text 
                               style = {{color: 'gray'}}>
                                   Create new Label
                           </Text>
                   </TouchableOpacity>}

                   <Appbar.Action
                        icon = {(this.state.labelTextBoxStatus) ? "check" : undefined}
                        onPress= {(this.state.labelTextBoxStatus) ? this.createLabel : null}
                    />
                </Appbar>
              
                <ScrollView>
                        {labelKey.length >0 ?
                        labelKey.reverse().map(key => (
                            <React.Fragment key={key}>
                                <Appbar style = {CreateNewLabelStyles.appbar_selected_level}>
                                    <Appbar.Action icon = {'label-outline'}/>
                                   
                                    <Text style = {CreateNewLabelStyles.appbar_selected_level_text}>
                                        {this.state.label_detail[key].label}
                                    </Text>

                                    <Appbar.Action
                                    icon = {require('../../assets/pen1.png')}/>
                                </Appbar>                                
                            </React.Fragment>
                        ))
                    : null}
                </ScrollView>
            </View>
        )
    }
}
