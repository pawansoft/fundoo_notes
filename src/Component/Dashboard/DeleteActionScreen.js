import React, { Component } from 'react';
import {
    ScrollView,
    Text,
    View
} from 'react-native';
import { Appbar, Button, Menu } from 'react-native-paper';
import RBSheet from 'react-native-raw-bottom-sheet';
import dashboardStyle from '../../Style/dashboardStyle';
import AsyncStorage from '@react-native-async-storage/async-storage';
import NotesHolderStyle from '../../Style/NotesHolderStyle';
import NoteServices from '../../../Services/firebase_services/NoteServices';

export class DeleteActionScreen extends Component {
    constructor(props) {
        super(props)
        this.state = {
            title: '',
            note: '',
            key: '',
            userid: '',
            isEmpty: false
        }
    }

    componentDidMount = async () => {
        const userId = await AsyncStorage.getItem('userId')
        await this.setState({
            userid: userId
        })

        if (this.props.route.params != undefined) {
            await this.setState({
                key: this.props.route.params.key,
                note: this.props.route.params.notes.NotesDetail.note,
                title: this.props.route.params.notes.NotesDetail.title
            })
        }
    }
    handleRBSheetOpenButton = async () => {
        this.RBSheet.open();
    }

    handleRestoreService = () => {
        NoteServices._restoreNoteService(this.state.userid, this.state.key, this.state.title, this.state.note)
            .then(() => this.props.navigation.push('Home', { screen: 'Delete' }))
            .catch(error => console.log(error))
    }

    handleDeleteNoteForEver = () => {
        NoteServices._deleteOneNoteService(this.state.userid, this.state.key)
            .then(() => this.props.navigation.push('Home', { screen: 'Delete' }))
            .catch(error => console.log(error))
    }

    render() {
        return (
            <View style={{ flex: 1 }}>
                <Appbar style={{ backgroundColor: 'white', width: '20%' }}>
                    <Appbar.Action
                        icon='keyboard-backspace'
                        onPress={() => this.props.navigation.push('Home', { screen: 'Delete' })}
                    />
                </Appbar>
                <ScrollView>
                    <View style={NotesHolderStyle.Title}>
                        <Text style={NotesHolderStyle.TitleFont}>
                            {this.state.title}
                        </Text>
                    </View>
                    <View style={NotesHolderStyle.Title}>
                        <Text>
                            {this.state.note}
                        </Text>
                    </View>
                </ScrollView>
                <View>
                    <Appbar style={{ backgroundColor: 'white', justifyContent: 'space-around' }}>
                        <Appbar.Action
                            icon='plus-box-outline' />
                        <Appbar.Action
                            title='Deleted note' />
                        <Appbar.Action
                            icon='dots-vertical'
                            onPress={this.handleRBSheetOpenButton} />
                    </Appbar>
                </View>
                <RBSheet
                    ref={ref => { this.RBSheet = ref }}
                    height={200}
                    customStyles={{
                        container: {
                            borderTopWidth: 1,
                            borderColor: "#d3d3d3",
                            marginBottom: 50
                        },
                        wrapper: {
                            backgroundColor: "transparent",
                        },
                    }}>
                    <View style={{ marginTop: '10%' }}>
                        <Menu.Item
                        icon = 'autorenew'
                            onPress={this.handleRestoreService}
                            title = ' Restore'/>
                        <Menu.Item
                            title = 'Delete forever'
                            icon = 'delete-outline'
                            onPress={this.handleDeleteNoteForEver}/>
                            
                    </View>
                </RBSheet>
            </View>
        )
    }
}