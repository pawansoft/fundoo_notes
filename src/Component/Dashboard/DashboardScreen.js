import React, { Component } from 'react';
import {
    View,
    ScrollView,
    Text,
} from 'react-native';
import BottomBar from './dashboardFooter';
import DashboardHeader from './DashboardHeader';
import NotesContainer from './NotesContainer';
import { Portal, Snackbar, Modal, Provider, Button, Appbar} from 'react-native-paper';
import Profile from './Profile';
import ProfileStyle from '../../Style/ProfileStyle';
import NoteServices from '../../../Services/firebase_services/NoteServices';

class DashboardScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            listView: true,
            showEmptyNoteSnackbar: false,
            showDeleteNoteSnackbar:false,
            showDelete: false,
            showProfileScreen: false
        }
    }

    showProfile = async () => {
        await this.setState({
            showProfileScreen: true
        })
    }

    hideProfile = async () => {
        await this.setState({
            showProfileScreen: false
        })
    }

    async componentDidMount() {

        if (this.props.route.params != undefined) {
            if (this.props.route.params.isEmpty != undefined) {
                await this.setState({
                    showEmptyNoteSnackbar: true
                })
            }
            else if(this.props.route.params.isDeleted != undefined){
                await this.setState({
                    showDeleteNoteSnackbar: true
                })
            }
        }
    }

    restoreNotesHandler = () =>{
        NoteServices._restoreNoteService(this.props.route.params.userid, 
                                            this.props.route.params.key, 
                                            this.props.route.params.title,
                                            this.props.route.params.note).then(() =>{
                                                this.props.navigation.push('Home', {screen : 'Notes'})
                                            }).catch(error => console.log(error))

    }

    snakbarHandler = async () => {
        await this.setState({
            showEmptyNoteSnackbar: false
        })
        this.props.navigation.setParams({ isEmptyNote: false })
    }

    deleteSnakbarHandler = async () => {
        await this.setState({
            showDeleteNoteSnackbar: false
        })
        this.props.navigation.setParams({ isDeleteNote: false })
    }

    selectView = async () => {
        await this.setState({
            listView: !this.state.listView
        })
    }


    render() {
        return (
            <Provider>
                <View style={{ flex: 1, justifyContent: "space-between" }}>
                    <View>
                        <DashboardHeader navigation={this.props.navigation} onPress={this.selectView} listView={this.state.listView} onSelectProfile={this.showProfile} />
                    </View>
                    <ScrollView>
                        <View>
                            <NotesContainer navigation={this.props.navigation} listview={this.state.listView} />
                        </View>
                    </ScrollView>
                    <View>
                        <BottomBar navigation={this.props.navigation} />
                    </View>
                    <View>
                        <Snackbar
                            style={{ marginBottom: '30%' }}
                            visible={this.state.showEmptyNoteSnackbar}
                            onDismiss={this.snakbarHandler}
                            duration={10000}>
                            Empty Note Discarded
                        </Snackbar>
                        <Snackbar
                        style={{ marginBottom: '30%', flexDirection: 'row', justifyContent: 'space-around', backgroundColor: 'grey'}}
                            visible={this.state.showDeleteNoteSnackbar}
                            onDismiss={this.snakbarHandler}
                            duration={10000}>
                            <View style = {{flexDirection: 'row', justifyContent: 'space-around'}}>
                            <View>
                                <Text style = {{marginTop: 10, color : 'white'}}>
                                    Note Deleted Successfully
                                </Text>
                            </View>
                            <View style = {{marginLeft: 50}}>
                                <Button
                                onPress = {this.restoreNotesHandler}>
                                    <Text style = {{color: '#cca300'}}>UNDO</Text>
                                </Button>
                            </View> 
                            </View>  
                        </Snackbar>
                        <Portal>
                            <Modal
                                visible={this.state.showProfileScreen}
                                onDismiss={this.hideProfile}
                                contentContainerStyle = {ProfileStyle.container}>
                                <Profile navigation={this.props.navigation} />
                            </Modal>
                        </Portal>
                    </View>
                </View>
            </Provider>
        )
    }
}

export default DashboardScreen;