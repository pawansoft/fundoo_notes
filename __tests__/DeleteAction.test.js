import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MockAsyncStorage from 'mock-async-storage';
import { DeleteActionScreen } from '../src/Component/Dashboard/DeleteActionScreen';
import { Appbar, Menu } from 'react-native-paper';
import NoteServices from '../Services/firebase_services/NoteServices';
import { AsyncStorage } from 'react-native';

const mockImpl = new MockAsyncStorage();
jest.mock('@react-native-async-storage/async-storage', () => mockImpl);

jest.mock('react-native-fetch-blob', () => {
  return {
    DocumentDir: () => {},
    polyfill: () => {},
  }
});

configure({adapter: new Adapter()})

jest.mock('react-native-localization', () => class RNLocalization {
    language = 'en-US'
  
    constructor (props) {
      this.props = props
      this.setLanguage(this.language)
    }
  
    setLanguage (interfaceLanguage) {
      this.language = interfaceLanguage
      if (this.props[interfaceLanguage]) {
        var localizedStrings = this.props[this.language]
        for (var key in localizedStrings) {
          if (localizedStrings.hasOwnProperty(key)) this[key] = localizedStrings[key]
        }
      }
    }
})

describe('Test Delete Action Functionality', () =>
{
    it('ProvideDeleteAction_WhenMatchecSnapshot_testShouldPass', () => {
        const component = shallow(<DeleteActionScreen/>)
        expect(component).toMatchSnapshot();
    })
    it('test all icons from DeleteActionScreen', () => {
        const component = shallow(<DeleteActionScreen/>)
        expect(component.find(Menu.Item).at(0).props().icon).toEqual('autorenew')
        expect(component.find(Menu.Item).at(1).props().icon).toEqual('delete-outline')
        expect(component.find(Appbar.Action).at(0).props().icon).toEqual('keyboard-backspace')
        expect(component.find(Appbar.Action).at(1).props().icon).toEqual('plus-box-outline')
        expect(component.find(Appbar.Action).at(3).props().icon).toEqual('dots-vertical')
    }) 
})

describe('Test functionality of delete screen', () => {
    it('PressRestoreButton_WhenNoteRestored_shouldNavigateToDelete', async() => {
        const navigation = { navigate: jest.fn()}
        const onPressEvent = jest.fn();

        const component = shallow(<DeleteActionScreen onPress = {onPressEvent} navigation = {navigation}/>)
        NoteServices._restoreNoteService()
        await component.instance().handleNewNoteCreateButton();
        expect(navigation.navigate).toBeCalledWith("NewNotes")
    })
})