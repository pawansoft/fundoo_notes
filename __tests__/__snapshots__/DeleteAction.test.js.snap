// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Delete Action Functionality ProvideDeleteAction_WhenMatchecSnapshot_testShouldPass 1`] = `ShallowWrapper {}`;
